TangoTest.cpp

```diff
@@ -1769,15 +1769,12 @@ void TangoTest::write_boolean_spectrum(Tango::WAttribute &attr)
 	attr.get_write_value(w_val);
 	/*----- PROTECTED REGION ID(TangoTest::write_boolean_spectrum) ENABLED START -----*/
 
-  const bool * p;
-  attr.get_write_value(p);
-
   long len = attr.get_write_value_length();
   DEBUG_STREAM << "Length :" << len << endl;
 
   len = (len <= kSpecLen) ? len : kSpecLen;
 
-  ::memcpy(attr_boolean_spectrum_read, p, len * sizeof(Tango::DevBoolean));
+  ::memcpy(attr_boolean_spectrum_read, w_val, len * sizeof(Tango::DevBoolean));
```

Comment:
Thanks for the PR. Can you clarify why `w_val` is correct and `p` not? Both were fetched via `attr.get_write_value`.

Reply:
The crash only occurred for the `long` (and potentially the `ulong`). data type It's a question of data size. The `long` type is 8 bytes long bits on x64 while Tango::DevLong is 4 bytes long on all plateforms. The `w_val` variable (who's part of the POGO generated code) is properly typed. A very old piece of code was overwriting 'w_val' 'using the ` long* p` variable to retrieve the write value. Because p is of the working type, it 's not properly set `attr.get_write_value` (i.e. `p` will be 0 is initialize to 0, undefined value otherwise). The crash used to occur during the call to `memcpy` where `p` was used a source address (invalid pointer). 

For any other data type, the code has been modified for consistency reason.

Reply:
Got it. This makes sense now.